src/odemis/driver/test/actuator_test.py

```diff
@@ -711,5 +713,396 @@ def test_fail_sensor(self):
+        f.result()
+        # check axis is referenced and children axes are referenced
+        self.assertTrue(self.dev.referenced.value[axis_name])
+        # self.assertTrue(self.child1.referenced.value[self.axis1])
+        # self.assertTrue(self.child1.referenced.value[self.axis2])
+
+    def tearDown(self):
+        axis_name = self.dev.axes.keys()[0]
+        self.dev.terminate()
+        super(TestCombinedFixedPostionActuator, self).tearDown()
+
+
+class TestRotationActuator(unittest.TestCase):
+
+    def setUp(self):
+
+        self.axis = "linear"
+        self.axis_name = "rz"
+
+        # # create 1 child
+        self.child1 = tmcm.TMCLController("rotstage1", "test", port="/dev/fake6",
+                                          axes=[self.axis], ustepsize=[3.392e-5],
+                                          unit=["rad"],
+                                          refproc="Standard",
+                                          )
+
+        self.dev_cycle = RotationActuator("stage", "stage", {self.axis_name: self.child1}, self.axis)
+
+    def test_unsupported_position(self):
+        """
+        test if unsupported position is handled correctly
+        """
+
+        axis_name = self.dev_cycle.axes.keys()[0]
+
+        # It's optional
+        if not hasattr(self.dev_cycle, "moveAbs"):
+            self.skipTest("Actuator doesn't support absolute move")
+
+        # generate random pos > 2pi
+        new_pos = random.uniform(2*math.pi, 10) + 0.0001  # to exclude 2pi
+        with self.assertRaises(ValueError):
+            f = self.dev_cycle.moveAbs({axis_name: new_pos})  # move
+            f.result()  # wait
+
+    def test_cycle_moveAbs(self):
+        """
+        test if any position is correctly reached
+        test if current position is requested nothing is done
+        """
+
+        axis_name = self.dev_cycle.axes.keys()[0]
+
+        # test don't change position
+        cur_pos = self.dev_cycle.position.value[axis_name]
+        f = self.dev_cycle.moveAbs({axis_name: cur_pos})
+        f.result()
+        self.assertEqual(self.dev_cycle.position.value[axis_name], cur_pos)
+
+        # test new position
+        new_pos = random.uniform(0, 2*math.pi)
+        f = self.dev_cycle.moveAbs({axis_name: new_pos})
+        f.result()
+        # check absolute difference is smaller half the ustepsize
+        self.assertLess(abs(self.dev_cycle.position.value[axis_name] - new_pos), self.child1._ustepsize[0]/2.)
+
+    def test_offset_moveAbs(self):
+        """
+        test if offset is correctly used
+        when accumulation of angles is overrunning 2pi (pos or neg) do referencing to zero
+        only works for cycle = 2pi
+        """
+
+        axis_name = self.dev_cycle.axes.keys()[0]
+
+        new_pos = 1.570796  # pi/2
+        # move 4*pi/2
+        for i in range(1, 5):
+            print i, "i"
+            f = self.dev_cycle.moveAbs({axis_name: new_pos*i})
+            f.result()
+            i += 1
```

Comment:
Normally you wouldn't increase `i` in a `for` loop. The statement actually does pretty much nothing as i is overwritten after every loop.

Reply:
Done